app/javascript/components/UpdateWork.jsx

```diff
@@ -0,0 +1,200 @@
+    console.log(this.props.work);
+    const route = APIRoutes.works.categories;
+    Requester.get(route).then(
+      response => {
+        this.setState({
+          categories: response,
+          componentDidMount: true
+        });
+        console.log(this.state.categories);
+      },
+      error => {
+        console.error(error);
+      }
+    );
+  }
+
+  handleChange(event) {
+    const work = this.state.work;
+    console.log(event.target.name);
+    console.log(event.target.value);
+    work[event.target.name] = event.target.value;
+    this.setState({ work: work });
+    console.log(this.state.work);
+  }
+
+  mediumHandleChange(event) {
+    const work = this.state.work;
+    work["medium"] = event.target.value;
+    console.log(event.target.value);
+    this.setState({ work: work });
+    console.log(this.state.work);
+  }
+
+  availabilityHandleChange(event) {
+    const work = this.state.work;
+    work["availability"] = event.target.value;
+    console.log(event.target.value);
+    this.setState({ work: work });
+    console.log(this.state.work);
+  }
+
+  handleSubmit(event) {
+
+    event.preventDefault();
+
+    let formData = new FormData();
+    formData.append('work[title]', this.state.work.title);
+    formData.append('work[material]', this.state.work.material);
+    formData.append('work[medium]', this.state.work.medium);
+    formData.append('work[availability]', this.state.work.availability);
+    formData.append('work[artist_id]', this.state.work.artist_id);
+
+    this.state.work.images.forEach((img) => {
+      console.log(img);
+      formData.append('work[attachments_attributes][]', img);
+    });
+
+    fetch(APIRoutes.works.update(this.state.work.id), {
+      method: 'PUT',
+      body: formData,
+      credentials: 'same-origin',
+      headers: {
+        "X_CSRF-Token": document.getElementsByName("csrf-token")[0].content
+      }
+    }).then((data) => {
+      window.location = `/artists/` + this.props.work.artist_id;
+    }).catch((data) => {
+      console.error(data);
+    });
+  }
+
+  onDrop(images) {
+    var currentImages = this.state.work.images;
+    const [newImages] = images;
+    currentImages.push(newImages);
+    console.log("ON DROP IMAGES-----------");
+    console.log(currentImages);
+
+    // Assign files dropped into component into state
+    var work = this.state.work;
+    work.images = currentImages;
+    this.setState({ work: work });
+  }
+
+  render() {
+    // TODO: Render thumbnails of images upon upload.
+    // TODO: Retrieve enums using works#get_work_category_enums method, rendering a dropdown
+    // rather than a text box below.
+    // Add error handling
+    // Also address the classNames. I think that they are part of the palantir CSS import,
+    // but we can't import this since it screws with our styles :(
+    return (
+      <div>
+        <form action={APIRoutes.works.update(this.state.work.id)} method='PUT' onSubmit={this.handleSubmit}>
```

Comment:
This seems like an exact duplicate of the `CreateWork` component. Instead, what we should do is pull out the form into a separate component, and populate the fields using props. Similarly, we should input props for the form action/method/submit callback.

Reply:
Then, within the `CreateWork` and `EditWork` components, simply create the form component and pass in the according props.

Reply:
fixed!